insights/combiners/cloud_provider.py

```diff
@@ -0,0 +1,170 @@
+"""
+Cloud Provider
+===============
+
+Combiner for Cloud information. It uses the results of
+:class:`InstalledRpms`,
+:class:`YumRepoList` and
+:class:`DMIDecode` parsers
+The combiner uses these parsers determine the Cloud Provider.
+
+Examples:
+    >>> cp.cp_bios_vendor
+    {'aws': 'amazon', 'google': '', 'azure': ''}
+    >>> cp.cp_bios_version
+    {'aws': '4.2.amazon', 'google': '', 'azure': ''}
+    >>> cp.cp_rpms
+    {'aws': ['rh-amazon-rhui-client-2.2.124-1.el7'], 'google': [], 'azure': []}
+    >>> cp.cp_yum
+    {'aws': [], 'google': [], 'azure': ['rhui-microsoft-azure-rhel7-2.2-74']}
+    >>> cp.cp_asset_tag
+    '7783-7084-3265-9085-8269-3286-77'
+    >>> cp.cp_uuid
+    'EC2F58AF-2DAD-C57E-88C0-A81CB6084290'
+
+"""
+
+from insights.core.plugins import combiner
+from insights.parsers.installed_rpms import InstalledRpms
+from insights.parsers.dmidecode import DMIDecode
+from insights.parsers.yum import YumRepoList
+from collections import namedtuple
+
+
+@combiner(optional=[InstalledRpms, DMIDecode, YumRepoList])
+class CloudProvider(object):
+    """
+    Combiner class to provide cloud vedor facts
```

Comment:
Let's include the different values that `cp.cloud_provider` can be so users will know which things they can test when using it. Or you might have them as constant module or class attributes or something (`CloudProvider.AWS`, `CloudProvider.GOOGLE`, etc.) so users can do `if cp.cloud_provider == CloudProvider.AWS)`.

(Also small typo with "vendor.")

Reply:
Thanks for the input Chris, made suggested changes.